### ISSUE TITLE
Exceptions during update of add-ons with dependents with optional Extensions

### ISSUE THREAD

**Issue by iress-ac** [zaproxy/zaproxy#7765]
### Describe the bug

This morning at 10:40 GMT we were using the zap2docker-stable image to successfully run zap scans, since about 12pm today ZAP has started to fail to find any urls in a file that hasn't changed since the previous successful run.

```
2023-03-03 12:52:08,736 Number of Imported URLs: 0
2023-03-03 12:52:08,736 Failed to import any URLs
Traceback (most recent call last):
  File "/zap/zap-api-scan.py", line 456, in main
    raise NoUrlsException()
NoUrlsException
```

### Steps to reproduce the behavior

1. Run a zap scan using the latest zap2docker image with an openapi spec with the following command:
`zap-api-scan.py -t ./api-spec.json -f 'openapi' -T 10 -c './config/zap.conf' -I -r 'zap_report.html'`
2. Observe the error above

### Expected behavior

The zap scan to complete successfully

### Software versions

`owasp/zap2docker-stable:latest`

### Errors from the zap.log file

[zap-log.txt](https://github.com/zaproxy/zaproxy/files/10882349/zap-log.txt)

### Would you like to help fix this issue?

- [ ] Yes

**Comment by psiinon** [comment]
@iress-ac we have identified an issue and re-released the stable docker image.
We believe this will fix this problem - can you confirm its ok for you now?

**Comment by iress-ac** [comment]
It's no better for us at the moment, we're still seeing the same errors.

**Comment by thc202** [comment]
Could you provide the log?

**Comment by psiinon** [comment]
And confirm you have pulled the latest image? 😁

**Comment by iress-ac** [comment]
I just swapped it to pulling the latest version explicitly by tag, `2.12.0` and it's working again! Thanks for the speedy resolution.

I'm curious - how did it spontaneously break for the last stable image?

**Comment by psiinon** [comment]
It looks like you have uncovered an issue in the core when updating some add-ons.
This is somewhat surprising as we have released these add-ons before without any problems.
We are investigating the core issue so that we can fix it in the next release and understand when it could re-occur.
The workaround (which we did this time) is to update the stable image with the latest add-ons - then theres no need to update them and the problem wont occur.
